### ISSUE TITLE
Question about pre-training

### ISSUE THREAD

**Issue by ZJUHJ** [wjun0830/CGDETR#8]
Thank you very much for releasing such a great work. I would like to ask if you will release the code and configuration for pre-training on the ego4d and video cc data sets?

**Comment by wjun0830** [comment]
We are not planning to release the code for the pertaining.

But simply replacing the univtg.py and transformer.py in the UniVTG repository with our model.py and transformer.py,
you may be able to prepare one yourself.

For the configuration, we enumerate below the texts in opt.json

'''
{
    "dset_type": "vlp",
    "dset_name": "vlp",
    "domain_name": null,
    "model_id": "cg_detr",
    "exp_id": "cgdetr_pt",
    "device": 0,
    "gpu_id": 0,
    "debug": false,
    "seed": 2018,
    "local_rank": -1,
    "eval_split_name": "val",
    "data_ratio": 1.0,
    "results_root": "results",
    "num_workers": 8,
    "no_pin_memory": false,
    "bsz": 64,
    "n_epoch": 10,
    "max_es_cnt": 200,
    "lr": 0.0001,
    "lr_drop": 200,
    "lr_gamma": 0.1,
    "lr_warmup": -1,
    "wd": 0.0001,
    "grad_clip": 0.1,
    "lw_saliency": 1.0,
    "lw_wattn": 1.0,
    "lw_ms_align": 1.0,
    "lw_distill": 1.0,
    "span_loss_coef": 10,
    "giou_loss_coef": 1,
    "label_loss_coef": 4,
    "contrastive_align_loss_coef": 0.0,
    "span_loss_type": "l1",
    "b_loss_coef": 10.0,
    "g_loss_coef": 1.0,
    "eos_coef": 0.1,
    "f_loss_coef": 10.0,
    "s_loss_intra_coef": 0.1,
    "s_loss_inter_coef": 0.1,
    "main_metric": "MR-full-R1@0.3-key",
    "eval_mode": null,
    "eval_bsz": 32,
    "eval_epoch": 100,
    "eval_init": false,
    "save_interval": 1,
    "resume": null,
    "resume_dir": null,
    "resume_all": false,
    "start_epoch": null,
    "no_sort_results": false,
    "max_before_nms": 1000,
    "max_after_nms": 10,
    "conf_thd": 0.0,
    "nms_thd": 0.7,
    "use_cache": 0,
    "max_q_l": 75,
    "max_v_l": 75,
    "clip_length": 2.0,
    "clip_len_list": null,
    "max_windows": 5,
    "add_easy_negative": 1,
    "easy_negative_only": 1,
    "round_multiple": 1,
    "train_path": [
        "data/ego4d/point_egoclip_wo_val.jsonl",
        "data/videocc/interval_900k.jsonl",
        "data/videocc/curve_5_window.jsonl"
    ],
    "eval_path": "data/qvhighlights_val.jsonl",
    "train_path_list": null,
    "eval_path_list": null,
    "feat_root_list": null,
    "no_norm_vfeat": false,
    "no_norm_tfeat": false,
    "v_feat_dirs": [
        "vid_slowfast",
        "vid_clip"
    ],
    "t_feat_dir": "txt_clip",
    "v_feat_dim": 2816,
    "t_feat_dim": 512,
    "ctx_mode": "video_tef",
    "v_feat_types": "slowfast_clip",
    "t_feat_type": "clip",
    "position_embedding": "sine",
    "n_input_proj": 2,
    "temperature": 0.07,
    "contrastive_align_loss": false,
    "contrastive_hdim": 64,
    "enc_layers": 3,
    "sub_enc_layers": 2,
    "dec_layers": 3,
    "dummy_layers": 2,
    "sent_layers": 1,
    "moment_layers": 1,
    "t2v_layers": 2,
    "dim_feedforward": 1024,
    "hidden_dim": 256,
    "input_dropout": 0.5,
    "dropout": 0.1,
    "droppath": 0.1,
    "txt_drop_ratio": 0,
    "use_txt_pos": false,
    "nheads": 8,
    "num_queries": 10,
    "num_dummies": 45,
    "total_prompts": 10,
    "num_prompts": 1,
    "pre_norm": false,
    "set_cost_span": 10,
    "set_cost_giou": 1,
    "set_cost_class": 4,
    "saliency_margin": 0.2,
    "aux_loss": true,
    "max_segment_num": 20,
    "max_frame_num": 200,
    "top_percent": 0.02,
    "qfvs_vid_feature": "fps1",
    "qfvs_txt_feature": "query",
    "qfvs_split": -1,
    "qfvs_dense_shot": -1,
    "qfvs_score_ensemble": -1,
    "qfvs_score_gather": -1,
    "qfvs_loss_gather": -1,
    "results_dir": "results/vlp-vlp/detr_pt-slowfast_clip-clip-45-2023_09_18_06_56_32"
}
'''
